database/rrdset.c

```diff
@@ -533,6 +533,7 @@ RRDSET *rrdset_create_custom(
 
     char fullid[RRD_ID_LENGTH_MAX + 1];
     snprintfz(fullid, RRD_ID_LENGTH_MAX, "%s.%s", type, id);
+    int changed_from_archived_to_active = 0;
```

Comment:
Considering that this variable is not used outside the the block `539 - 667`, we could move the declaration inside the conditional. There is also a code quality analysis that we use that in normal situation does this check.